cypress/integration/creatorOnboardingFlows/creatorSignup.spec.js

```diff
@@ -0,0 +1,100 @@
+describe('Creator Signup Page', () => {
+  beforeEach(() => {
+    cy.testSetup();
+    cy.visit('/enter?state=new-user');
+  });
+
+  it('should display a welcome message', () => {
+    cy.findByText("Let's start your Forem journey!").should('be.visible');
+  });
+
+  it('should display instructions', () => {
+    cy.findByText('Create your admin account first.').should('be.visible');
+    cy.findByText("Then we'll walk you through your Forem setup.").should(
+      'be.visible',
+    );
+  });
+
+  it('should display a validated username hint that correlates to the name entered', () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm')
+      .findByText(/^Name$/)
+      .type('1 Forem creator name! Also test a maximum length string');
+
+    // restricts the string to 20 characters, lowercased,
+    // replaces spaces and special_characters with an underscore
+    cy.contains('1_forem_creator_name__also_tes');
+  });
+
+  it('should show and focus on the username field when clicking on the edit icon', () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm')
+      .findByText(/^Name$/)
+      .type('Forem creator name');
+
+    cy.get('.js-creator-edit-username').click();
```

Comment:
These tests are awesome, but if we wanted to optimise we could replace a lot of classname and text selectors with `findByRole...` and `findByLabelText...` - those queries tend to be more robust and have the added benefit of testing the accessibility of the UI & closer to real user interactions than implementation details

Reply:
Oooh thanks @aitchiss this makes sense, I'll look into this!